api/src/wfl/module/covid.clj

```diff
@@ -14,19 +14,32 @@
             [wfl.service.rawls :as rawls]
             [wfl.util :as util]
             [wfl.wfl :as wfl])
-  (:import (java.time OffsetDateTime)))
+  (:import [java.time OffsetDateTime]))
 
 (def pipeline "Sarscov2IlluminaFull")
 
+(defn row-ids-yesterday
+  "Return the row IDs added to TABLE in DATASET yesterday."
+  [dataset table]
+  (-> dataset
+      (datarepo/query-table-between table (util/days -1) [:datarepo_row_id])
+      :rows flatten))
+
 ;; TODO: implement COVID workload creation
 ;;  - make sure permissions/inputs are right upfront
 ;;  - dispatch on source/sink/executor
 ;;  - store information into database
 (defn create-covid-workload!
   [tx {:keys [source sink executor] :as request}])
 
+(defn ^:private get-snapshots-from-workspace
+  [workspace])
+
 (defn start-covid-workload!
-  [])
+  "Mark WORKLOAD with a started timestamp."
+  [tx {:keys [id started] :as workload}]
+  (jdbc/update! tx :workload {:started (OffsetDateTime/now)} ["id = ?" id])
+  (workloads/load-workload-for-id tx id))
```

Comment:
Minor optimisation here, do we need to re-load the workload or can we just update the `:started` field?

Reply:
`start-` things tend to return the workload, but we don't have to, of course.